client/src/game/ui/notes.ts

```diff
@@ -75,7 +75,7 @@ export const set = (order: number, note: string) => {
 const getNoteKeywords = (
   note: string,
 ) => {
-  const regexp = /\[([^\]]*)\]|\|([^|[]*)$|^([^|]+)$/g;
+  const regexp = /\[(.*?)\]|\|([^[|]*$)|(^[^[|]+$)/g;
```

Comment:
can you add a comment explaining what the regex is doing piece by piece

Reply:
does what I added help?

Reply:
perfect, thank you very much

i always like to write healthy comments next to any non-trivial code like this